src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1468,6 +1468,27 @@ public static int Sign(float value)
             throw new ArithmeticException(SR.Arithmetic_NaN);
         }
 
+        // Helper functions for CopySign
+        // >= 0: returns 1
+        // < 0: returns -1
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int SignZeroToOne(int value)
+        {
+            return (value >> (32 - 2)) | 1;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int SignZeroToOne(nint value)
+        {
+            return (int)(value >> (8 * nint.Size - 2)) | 1;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int SignZeroToOne(long value)
+        {
+            return (int)(value >> (64 - 2)) | 1;
+        }
```

Comment:
Casting to int32 will introduce unnecessary truncate&extend on 64 bit platform, but saves a move on 32 bit platform.

Also note that we typically put these methods in System.Numerics.BitOperations.

Reply:
I think returning an int (Int32) for Sign is reasonable, because it is the common return type for sign functions. Also, using a shorter bit length than long (Int64) may provide optimization opportunities for subsequent multiplications. Regarding where to put these methods, I think Sign and Abs are closely related, so it makes sense to group them together. Besides, the original code already referenced some throw helper methods from Math, so moving them to System.Numerics.BitOperations would make the code more scattered.

Reply:
In such primitive function, one instruction counts. The helper isn't exposed to public, so it should be most performant form. In case when needed, you can use `#if TARGETS_64BIT` to provide different implementation.

We need to understand the performance better.